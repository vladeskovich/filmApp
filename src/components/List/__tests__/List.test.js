import React from 'react';
import List from '../List';

describe('List', () => {
  it('should render List component ', () => {
    const props = {
      data: [
        { id: 1, name: 'Action' },
      ],
      className: 'test-list',
      direction: 'vertical',
      itemClassName: 'test-list',
      ordered: true,
    };

    const component = shallow(<List {...props}>{(element) => (
      <div>{element.name}</div>
    )
    }
    </List>);

    expect(component).toMatchSnapshot();
  });

  it('should render List component ', () => {
    const props = {
      data: [
        { name: 'Action' },
      ],
      className: 'test-list',
      direction: 'vertical',
      itemClassName: 'test-list',
      ordered: false,
    };

    const component = mount(<List {...props}>{(element) => (
      <div>{element.name}</div>
    )
    }
    </List>);

    expect(component.find('ul').length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
