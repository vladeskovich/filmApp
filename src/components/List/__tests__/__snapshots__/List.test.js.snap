// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render List component  1`] = `
<ol
  className="List__list_3qljzsty test-list List__vertical_2QU82bDP"
>
  <li
    className="test-list"
    key="1"
  >
    <div>
      Action
    </div>
  </li>
</ol>
`;

exports[`List should render List component  2`] = `
<List
  className="test-list"
  data={
    Array [
      Object {
        "name": "Action",
      },
    ]
  }
  direction="vertical"
  itemClassName="test-list"
  ordered={false}
>
  <ul
    className="List__list_3qljzsty test-list List__vertical_2QU82bDP"
  >
    <li
      className="test-list"
      key="0"
    >
      <div>
        Action
      </div>
    </li>
  </ul>
</List>
`;

exports[`List should render List component without props 1`] = `""`;
